### ISSUE TITLE
libwasmer_darwin_amd64.dylib - no such file

### ISSUE THREAD

**Issue by ctindogarus4f** [ElrondNetwork/elrond-sdk-erdpy#122]
On macOS, `libwasmer_darwin_amd64.dylib` is not installed when installing `erdpy` so `erdpy contract test` fails.

**OS used:**
`macOS Monterey 12.2.1`

**`erdpy` installed via:**
```bash
wget -O erdpy-up.py https://raw.githubusercontent.com/ElrondNetwork/elrond-sdk-erdpy/master/erdpy-up.py
python3.8 erdpy-up.py
```

**Error:**
```bash
$ erdpy contract test
INFO:projects.core:run_tests.project: /Users/constantindogaru/elrond-s4f/elrond-smart-contracts/vesting
INFO:myprocess:run_process: ['/Users/constantindogaru/elrondsdk/vmtools/mandos-test', '/Users/constantindogaru/elrond-s4f/elrond-smart-contracts/vesting/mandos'], in folder: None
CRITICAL:cli:External process error:
Command line: ['/Users/constantindogaru/elrondsdk/vmtools/mandos-test', '/Users/constantindogaru/elrond-s4f/elrond-smart-contracts/vesting/mandos']
Output: dyld[38511]: Library not loaded: @executable_path/libwasmer_darwin_amd64.dylib
  Referenced from: /Users/constantindogaru/elrondsdk/vmtools/v1.4.44/wasm-vm-1.4.44/cmd/test/test
  Reason: tried: '/Users/constantindogaru/elrondsdk/vmtools/v1.4.44/wasm-vm-1.4.44/cmd/test/libwasmer_darwin_amd64.dylib' (no such file), '/usr/local/lib/libwasmer_darwin_amd64.dylib' (no such file), '/usr/lib/libwasmer_darwin_amd64.dylib' (no such file)
```

To solve this issue, I have to manually copy https://github.com/ElrondNetwork/wasm-vm/blob/master/wasmer/libwasmer_darwin_amd64.dylib inside `'/usr/local/lib` which is not ideal and I'm pretty sure it blocks many users. Can we please get erdpy installation fixed, so `libwasmer_darwin_amd64.dylib` gets installed by default when installing `erdpy`?

**Comment by andreibancioiu** [comment]
@ctindogarus4f. thank you for the detailed description :rocket: 

Currently, our Github Workflow builds uses `macOS 11.6.4`, and the tests are passing: https://github.com/ElrondNetwork/elrond-sdk-erdpy/actions/workflows/erdpy.yml.

It is likely (unconfirmed yet) that in more recent macOS versions (e.g. _monterey_), when running a file that sym-links to an executable and trying to resolve dynamic libraries referenced using `@executable_path/my-library.dylib`, the system does not search the library near the symlink anymore, but only near the target executable (in our case, `mandos-test` is the symlink, while `v1.4.44/wasm-vm-1.4.44/cmd/test/test` is the executable).

Normally, copying the .dylib file in `v1.4.44/wasm-vm-1.4.44/cmd/test` should be the preferred workaround - copying to `/usr/local/lib` is not recommended, actually.

A fix is on the work. Thank you :pray:

**Comment by ctindogaru** [comment]
Amazing, thank you for looking into it so fast Andrei 👍 

Feel free to close the issue when the fixed version of `erdpy` comes up.

**Comment by andreibancioiu** [comment]
@ctindogarus4f, we have released the fix in **v1.2.3**. We will keep the issue open for a few more days, and then close it if everything is all right :pray:

**Comment by ctindogaru** [comment]
Just re-installed `erdpy` now. Everything works like a charm!

Thanks once again for dealing with this. Issue can be closed even now from my perspective 💯 🥇
